node_cli/core/ssl/check.py

```diff
@@ -201,8 +201,15 @@ def check_ssl_connection(host, port, silent=False):
     ]
     expose_output = not silent
     with detached_subprocess(ssl_check_cmd, expose_output=expose_output) as dp:
-        time.sleep(1)
-        code = dp.poll()
-        if code is not None:
-            logger.error('Healthcheck connection failed')
-            raise SSLHealthcheckError('OpenSSL connection verification failed')
+        timeout = 20
+        try:
+            dp.wait(timeout=timeout)
```

Comment:
This consistently times out :/

Based on https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait I suspect:

> Note This will deadlock when using stdout=PIPE or stderr=PIPE and the child process generates enough output to a pipe such that it blocks waiting for the OS pipe buffer to accept more data. Use [Popen.communicate()](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate) when using pipes to avoid that.

Reply:
Reading now https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate

> [communicate()](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate) returns a tuple (stdout_data, stderr_data).

and looking how `def detached_subprocess()` is defined, that it `p.stdout.read()` and grabs the output, I am inclined to actually lave the loop and `dp.poll()` as it was.

Reply:
With `dp.wait()` in consistently throws `subprocess.TimeoutExpired` and `openssl s_client -connect 127.0.0.1:4536 -verify_return_error -verify 2` is not finishing, but it is running.

Reply:
Hm.. it times out for me now with `dp.poll()` as well, but it worked before :thinking: